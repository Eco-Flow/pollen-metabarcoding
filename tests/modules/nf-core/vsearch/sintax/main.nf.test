nextflow_process {

    name "Test Process VSEARCH_SINTAX"
    script "modules/nf-core/vsearch/sintax/main.nf"
    process "VSEARCH_SINTAX"

    test("Should run without failures") {

        when {
            params {
                clean = false
                sintax_strand = "both"
                sintax_cutoff = 0.95
            }
            process {
                """
                input[0] = Channel.of( [ [ "id":"ERR2537816", "single_end": false ], file("${projectDir}/data/sintax-input/ERR2537816.derep.fasta") ] )
                input[1] = file("s3://pollen-metabarcoding-test-data/data/viridiplantae_all_2014.sintax.fa")
                """
            }
        }

        then {
            assert process.success
            with(process.out) {
              assert tsv.size() == 1
              assert tsv[0].size() == 2
              assert tsv[0][0].size() == 2
            }

        }

    }

}
