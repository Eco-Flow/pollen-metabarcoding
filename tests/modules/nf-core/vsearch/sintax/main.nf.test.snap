{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "ERR2537816",
                            "single_end": false
                        },
                        "ERR2537816.tsv:md5,84351a56e4ddf56df7403c9ffefa0f1b"
                    ]
                ],
                "1": [
                    "versions.yml:md5,6584e256329124c6abe9e76715edfe92"
                ],
                "tsv": [
                    [
                        {
                            "id": "ERR2537816",
                            "single_end": false
                        },
                        "ERR2537816.tsv:md5,84351a56e4ddf56df7403c9ffefa0f1b"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,6584e256329124c6abe9e76715edfe92"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-04T12:52:09.921597141"
    }
}